hw-js-7/hw-js-7-2/script.js

```diff
@@ -1,24 +1,51 @@
 "use strict";
-document.getElementsByClassName('button')[0].addEventListener('click', open);
-function open(){
-  alert('Very interesting homework for me.');
-}
 
-//next window
-let modal = document.getElementById('myModal');
-let btn = document.getElementById("myBtn");
-let closed = document.getElementsByClassName("close")[0];
+class Marker {
+  constructor(c, p){
+    this.c = c;
+    this.p = p;
+  }
 
-btn.onclick = function() {
-  modal.style.display = "block";
-}
+  get markerProps(){
+    return [this.c, this.p];
+  }
+
+  set markerProps(newProps){
+    [this.c, this.p] = [...newProps];
+  }
 
-closed.onclick = function() {
-  modal.style.display = "none";
+  print(line){
+    let t = document.getElementById("content");
+    for (let i = 0; i < line.length; i++){
+      if (this.p != 0){
+        if (line[i] == " "){
+          this.p += 0.5;
+        }
+        t.innerHTML += line[i];
+        t.style.color = this.c;
+        this.p -= 0.5;
+      } else {
+        document.write("Marker is over!");
+        break;
+      }
+    }
+  }
 }
 
-window.onclick = function(event) {
-  if (event.target == modal) {
-    modal.style.display = "none";
+class FilledMarker extends Marker {
+  fill(p){
+    if (p > 100){
+      p = 100;
+    } else {
+      this.p += p;
+    }
   }
-}
\ No newline at end of file
+}
+
+let marker = new FilledMarker("#191970", 5);
+marker.fill(31);
+
+const l = 'Implement a class that describes a simple marker. The class should have the following components: field that stores the color of the marker; a field that stores the amount of ink in the marker (in percent); Implement a refueling marker class by.';
+marker.print(l);
+
+document.body.setAttribute("style", "font-size:18px; text-align:center; font-family: cursive; margin-bottom:5%;");
```

Comment:
That type of coding is hard to maintain. Imagine someone from your team tries to change some styles and he can't find where it is. Make your code easier to understand and simple to maintain